"Adicionado impressão de mdf-e by: marcosgerene"

robertorp opened the pull request:
Impressão de MDF-e implementada por @marcosgerene 
Somente fiz uns testes e pull

robertorp committed:
"Adicionado impressão de mdf-e by: marcosgerene"

robertorp committed:
"alterado projeto para versão 4.0 do .net"

marcosgerene commented:
O que muda da 4.0 para 4.5.2?

Nos meus projetos uso a 4.6.1 e ele exige win 7sp1, win 8 sp1 ou win10.

A 4.0 permite win7 sem o sp1? Até qual versão permite o XP?

robertorp commented:
Nos meus projetos uso a 4.5.2. Olha eu acho que muda no jeito de criar Task, se não me engano o 4.0 não tem Task.Run e com toda certa mais coisas... Mas não sei especificar as diferenças para você.

Pesquisando aqui tem isso http://stackoverflow.com/questions/10492647/differences-between-net-4-0-and-net-4-5-in-high-level-in-net

rftd commented:
Versão 4.0 para baixo para ser usado no XP e já não recomendo pois este ano vem a NFe 4.0 ai tem que ser no minimo .net framework 4.5 por causa do TLS 1.2

Mais informações.
https://msdn.microsoft.com/pt-br/library/8z6watww(v=vs.110).aspx

rftd commented:
no 4.0 não tem os async e await precisa de lib do nuget para isso a task muda so um pouco ficando
Task.Factory.StartNew o resto é igual.

Para keyword asyn e await
https://www.nuget.org/packages/Microsoft.Bcl/

marcosgerene commented:
Eu atualmente uso a 4.6.1 e C# 6.0 mas vira e mexe esbarro num windows 7 sem SP1 que da uma quebrada nas minhas pernas, a 4.5 roda sem o SP1 do windows 7?

Existe diferença gritante de performace entre 4.6.1 e 4.5?

PS: Meu software é em winforms

robertorp committed:
adicionado classe configuracaocertificado no dfe.util

robertorp committed:
Unicifaco classes de certificado digital
Correção na dhemi e dhevento do mdf-e pois os mesmos não tem utc
manual mdf-e pagina 60 dhevento
pagina 99 dhemi
observação: não estava passando na validação dos schemas
Adicionado cache de certificado digital no mdf-e

robertorp commented:
Adicionado unificação de classes de certificado digital.
Feito correção do commit anterior, no commit anterior colocaram utc na dhemi e dhevento no mdf-e, não tem na mdf-e, manual 1.0 paginas 60 e 99 
Feito cache do certificado digital no mdf-e (com as mesmas classes do nf-e, simplesmente passei elas para o projeto dfe.util e delete as que tinha antes)
alterado appteste mdf-e para ter a opção de cache
